src/attributes.jl

```diff
@@ -23,6 +23,26 @@ abstract type AbstractConstraintAttribute end
 
 const AnyAttribute = Union{AbstractInstanceAttribute, AbstractVariableAttribute, AbstractConstraintAttribute}
 
+"""
+    supportsattribute(instance::AbstractInstance, attr::AbstractInstanceAttribute)::Bool
+
+Return a `Bool` indicating whether `instance` supports the instance attribute `attr`.
+
+    supportsattribute(instance::AbstractInstance, attr::AbstractVariableAttribute, ::Type{VariableIndex})::Bool
+
+Return a `Bool` indicating whether `instance` supports the variable attribute `attr`.
+
+    supportsattribute(instance::AbstractInstance, attr::AbstractConstraintAttribute, ::Type{ConstraintIndex{F,S}})::Bool where {F,S}
+
+Return a `Bool` indicating whether `instance` supports the constraint attribute `attr`.
+
+In other words, it should return `true` if `copy!(instance, src)` does not return `CopyUnsupportedAttribute` when the attribute `attr` is set to `src`.
+If the attribute is only not supported in specific circumstances, it should still return `true`.
+"""
+function supportsattribute end
```

Comment:
Would it be consistent with `set` and `get!` to name this `supports`?

Reply:
Yes, it would